qcodes_contrib_drivers/drivers/SingleQuantum/SingleQuantum.py

```diff
@@ -0,0 +1,498 @@
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+
+import numpy as np
+import socket
+import json
+import threading
+import time
+import sys
+
+from qcodes.instrument.ip import IPInstrument
+from qcodes.instrument.base import Parameter
+from qcodes.instrument.parameter import ParameterWithSetpoints, MultiParameter
+from qcodes.utils.validators import Arrays
+
+
+def synchronized_method(method, *args, **kws):
+    outer_lock = threading.Lock()
+    lock_name = "__" + method.__name__ + "_lock" + "__"
+
+    def sync_method(self, *args, **kws):
+        with outer_lock:
+            if not hasattr(self, lock_name):
+                setattr(self, lock_name, threading.Lock())
+            lock = getattr(self, lock_name)
+            with lock:
+                return method(self, *args, **kws)
+    sync_method.__name__ = method.__name__
+    sync_method.__doc__ = method.__doc__
+    sync_method.__module__ = method.__module__
+    return sync_method
+
+
+def synchronized_with_attr(lock_name):
+    def decorator(method):
+        def synced_method(self, *args, **kws):
+            lock = getattr(self, lock_name)
+            with lock:
+                return method(self, *args, **kws)
+        synced_method.__name__ = method.__name__
+        synced_method.__doc__ = method.__doc__
+        synced_method.__module__ = method.__module__
+        return synced_method
+    return decorator
+
+
+class SQTalk(threading.Thread):
+    def __init__(self, TCP_IP_ADR='localhost', TCP_IP_PORT=12000,
+                 error_callback=None):
+        threading.Thread.__init__(self)
+        self.TCP_IP_ADR = TCP_IP_ADR
+        self.TCP_IP_PORT = TCP_IP_PORT
+
+        self.socket = socket.socket(
+            socket.AF_INET, socket.SOCK_STREAM)
+        self.socket.connect((self.TCP_IP_ADR, self.TCP_IP_PORT))
+        self.socket.settimeout(.1)
+        self.BUFFER = 10000000
+        self.shutdown = False
+        self.labelProps = dict()
+
+        self.error_callback = error_callback
+
+        self.lock = threading.Lock()
+
+    @synchronized_method
+    def close(self):
+        # Print("Closing Socket")
+        self.socket.close()
+        self.shutdown = True
+
+    @synchronized_method
+    def send(self, msg):
+        if sys.version_info.major == 3:
+            self.socket.send(bytes(msg, "utf-8"))
+        if sys.version_info.major == 2:
```

Comment:
QCoDeS does not support python 2 so from that point of view this is probably not needed

Reply:
Removed the check